src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala

```diff
@@ -0,0 +1,72 @@
+package com.kinja.amqp
+
+import com.kinja.amqp.model.Message
+import com.kinja.amqp.model.MessageConfirmation
+
+import org.slf4j.{ Logger => Slf4jLogger }
+
+import scala.util.Failure
+import scala.util.Success
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object UnconfirmedMessageRepeater {
```

Comment:
that is what we do with indexer sentinels too in the current system, and this resend loop is basically the replacement for sentinels, so akka scheduler is perfect here.